test/e2e-test/app_embed_rollout_test.go

```diff
@@ -0,0 +1,358 @@
+		By("Verify  pod status")
+		Eventually(func() error {
+			podList := corev1.PodList{}
+			if err := k8sClient.List(ctx, &podList, client.MatchingLabels(kc.Spec.Template.Labels), client.InNamespace(namespaceName)); err != nil {
+				return err
+			}
+			if len(podList.Items) != int(*kc.Spec.Replicas) {
+				return fmt.Errorf("pod number error")
+			}
+			for _, pod := range podList.Items {
+				if pod.Status.Phase != corev1.PodRunning {
+					return fmt.Errorf("pod status error %s", pod.Status.Phase)
+				}
+				if pod.Spec.Containers[0].Command[1] != fmt.Sprintf("stress-cpu=%s", cpu) {
+					return fmt.Errorf("pod cmmond haven't updated")
+				}
+			}
+			return nil
+		}, time.Second*120, time.Microsecond).Should(BeNil())
+	}
+
+	updateAppWithCpuAndPlan := func(app *v1beta1.Application, cpu string, plan *v1alpha1.RolloutPlan) {
+		Eventually(func() error {
+			checkApp := new(v1beta1.Application)
+			if err := k8sClient.Get(ctx, ctypes.NamespacedName{Namespace: namespaceName, Name: app.Name}, checkApp); err != nil {
+				return err
+			}
+			updateProperty := fmt.Sprintf(`{"cmd":["./podinfo","stress-cpu=%s"],"image":"stefanprodan/podinfo:4.0.3","port":8080,"replicas":6}`, cpu)
+			checkApp.Spec.Components[0].Properties.Raw = []byte(updateProperty)
+			checkApp.Spec.RolloutPlan = plan
+			if err := k8sClient.Update(ctx, checkApp); err != nil {
+				return err
+			}
+			return nil
+		}, time.Second*30, time.Microsecond*300).Should(BeNil())
+	}
+
+	It("Test upgrade application", func() {
+		plan := &v1alpha1.RolloutPlan{
+			RolloutStrategy: v1alpha1.IncreaseFirstRolloutStrategyType,
+			RolloutBatches: []v1alpha1.RolloutBatch{
+				{
+					Replicas: intstr.FromString("50%"),
+				},
+				{
+					Replicas: intstr.FromString("50%"),
+				},
+			},
+			TargetSize: pointer.Int32Ptr(6),
+		}
+		appName = "app-rollout-1"
+		app := generateNewApp(appName, namespaceName, "clonesetservice", plan)
+		Expect(k8sClient.Create(ctx, app)).Should(BeNil())
+		verifyRolloutSucceeded(utils.ConstructRevisionName(appName, 1), "1")
+		updateAppWithCpuAndPlan(app, "2", plan)
+		verifyRolloutSucceeded(utils.ConstructRevisionName(appName, 2), "2")
+		updateAppWithCpuAndPlan(app, "3", plan)
+		verifyRolloutSucceeded(utils.ConstructRevisionName(appName, 3), "3")
+	})
+
+	It("Test application only upgrade batchPartition", func() {
+		plan := &v1alpha1.RolloutPlan{
+			RolloutStrategy: v1alpha1.IncreaseFirstRolloutStrategyType,
+			RolloutBatches: []v1alpha1.RolloutBatch{
+				{
+					Replicas: intstr.FromString("50%"),
+				},
+				{
+					Replicas: intstr.FromString("50%"),
+				},
+			},
+			TargetSize: pointer.Int32Ptr(6),
+		}
+		appName = "app-roll-out-2"
+		app := generateNewApp(appName, namespaceName, "clonesetservice", plan)
+		Expect(k8sClient.Create(ctx, app)).Should(BeNil())
+		verifyRolloutSucceeded(utils.ConstructRevisionName(appName, 1), "1")
+		app.Spec.RolloutPlan.BatchPartition = pointer.Int32Ptr(0)
```

Comment:
why update the batchPartition outside of the plan when you are going to set it again immediately (line 278)